web_translate_dialog/static/src/js/web_translate_dialog.js

```diff
@@ -0,0 +1,224 @@
+/* Copyright 2012 Guewen Baconnier (Camptocamp SA)
+   Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_translate_dialog.translate_dialog', function(require){
+"use strict";
+
+var core = require('web.core');
+var BasicController = require('web.BasicController');
+var data = require('web.data');
+var Context = require('web.Context');
+var concurrency = require('web.concurrency');
+var Dialog = require('web.Dialog');
+var FormView = require('web.FormView');
+var View = require('web.AbstractView');
+var session  = require('web.session');
+var rpc = require('web.rpc');
+var dom = require('web.dom');
+
+var _t = core._t;
+var QWeb = core.qweb;
+var Mutex = concurrency.Mutex;
+
+var translateDialog = Dialog.extend({
+    template: "TranslateDialog",
+    init: function(parent, options) {
+        this._super(parent,
+                    {title: _t("Translations"),
+                     width: '90%',
+                     height: '80%'}
+                    );
+        this.view_language = session.user_context.lang;
+        this.view = parent;
+        this.view_type = parent.viewType || '';
+        this.$view_form = null;
+        this.$sidebar_form = null;
+        this.translatable_field = options.field.fieldName;
+        this.res_id = options.res_id;
+        this.languages = null;
+        this.languages_loaded = $.Deferred();
+        (new data.DataSetSearch(this, 'res.lang', parent.searchView.dataset.get_context(),
+                                [['translatable', '=', '1']])).read_slice(['code', 'name'],
+                                { sort: 'id' }).then(this.on_languages_loaded);
+    },
+    on_languages_loaded: function(langs) {
+        this.languages = langs;
+        this.languages_loaded.resolve();
+    },
+    open: function() {
+        // the template needs the languages
+        return $.when(this.languages_loaded).then($.proxy(this._super, this));
+    },
+    start: function() {
+        var self = this;
+        this.$el.find('.oe_translation_field').change(function() {
```

Comment:
All widgets have "$" method that call to 'this.$el.find'. You can replace all ```this.$el.find(``` by ```this.$(```